bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/StreamServerGroupHandler.java

```diff
@@ -0,0 +1,235 @@
+            if (msg instanceof HttpRequest) {
+                HttpRequest httpRequest = (HttpRequest) msg;
+                String requestIP = "("
+                        + ((InetSocketAddress) ctx.channel().remoteAddress()).getAddress().getHostAddress() + ")";
+                if (!whiteList.contains(requestIP) && !whiteList.equals("DISABLE")) {
+                    logger.warn("The request made from {} was not in the whitelist and will be ignored.", requestIP);
+                    return;
+                } else if (HttpMethod.GET.equals(httpRequest.method())) {
+                    // Some browsers send a query string after the path when refreshing a picture.
+                    QueryStringDecoder queryStringDecoder = new QueryStringDecoder(httpRequest.uri());
+                    switch (queryStringDecoder.path()) {
+                        case "/ipcamera.m3u8":
+                            if (ipCameraGroupHandler.hlsTurnedOn) {
+                                String debugMe = ipCameraGroupHandler.getPlayList();
+                                logger.debug("playlist is:{}", debugMe);
+                                sendString(ctx, debugMe, "application/x-mpegurl");
+                                return;
+                            } else {
+                                logger.warn(
+                                        "HLS requires the groups startStream channel to be turned on first. Just starting it now.");
+                                String channelPrefix = "ipcamera:" + ipCameraGroupHandler.getThing().getThingTypeUID()
+                                        + ":" + ipCameraGroupHandler.getThing().getUID().getId() + ":";
+                                ipCameraGroupHandler.handleCommand(new ChannelUID(channelPrefix + CHANNEL_START_STREAM),
+                                        OnOffType.ON);
+                            }
+                            break;
+                        case "/ipcamera.jpg":
+                            sendSnapshotImage(ctx, "image/jpg");
+                            return;
+                        default:
+                            if (httpRequest.uri().contains(".ts")) {
+                                sendFile(ctx, resolveIndexToPath(httpRequest.uri()) + httpRequest.uri().substring(2),
+                                        "video/MP2T");
+                            } else if (httpRequest.uri().contains(".jpg")) {
+                                sendFile(ctx, httpRequest.uri(), "image/jpg");
+                            } else if (httpRequest.uri().contains(".m4s") || httpRequest.uri().contains(".mp4")) {
+                                sendFile(ctx, httpRequest.uri(), "video/mp4");
+                            }
+                    }
+                }
+            }
+        } finally {
+            ReferenceCountUtil.release(msg);
+        }
+    }
+
+    private void sendSnapshotImage(ChannelHandlerContext ctx, String contentType) throws IOException {
```

Comment:
This exception is never thrown.
```suggestion
    private void sendSnapshotImage(ChannelHandlerContext ctx, String contentType) {
```